include/shell/shell_ipc.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef SHELL_IPC_H_
+#define SHELL_IPC_H_
+
+#include <kernel.h>
+#include <ipc/ipc_service.h>
+#include <shell/shell.h>
+#include <sys/ring_buffer.h>
```

Comment:
Limit the number of unnecessary includes. If you need them in the implementation unit then add them there

```suggestion
#include <shell/shell.h>
```